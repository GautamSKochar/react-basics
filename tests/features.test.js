import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import Featured from '../components/Features';

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

const raf = global.requestAnimationFrame = (cb) => {
    setTimeout(cb, 0)
}
configure({ adapter: new Adapter() });

test('Feature Component Should render as expected', () => {
    const key = 1;
    const value = 'newww text';
    const dummyFun = function () {
        console.log("dummy method called");
    }
    const component = shallow(<Featured key={key} index={1}
        updateCommentText={dummyFun}
        deleteFromBoard={dummyFun}>{value}</Featured>);

    const tree = toJson(component);
    expect(tree).toMatchSnapshot();

});

test('onEdit click event', () => {
    const dummyFun = function () {
        console.log("dummy method called");
    }
    const component = mount(<Featured key={1} index={1}
        updateCommentText={dummyFun}
        deleteFromBoard={dummyFun}>{"value"}</Featured>);
    // console.log("cmpt is:",toJson(component));
    const btn = component.find('#editBtn');
    console.log("btn props are:", btn.props());
    console.log(component.state());    
    btn.simulate('click');
    console.log(component.state());
    expect(component.state().editing).toEqual(true);
});