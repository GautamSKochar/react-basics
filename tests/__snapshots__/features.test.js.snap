// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Feature Component Should render as expected 1`] = `
<div
  className="container"
  style={
    Object {
      "display": "inline",
      "float": "left",
      "maxWidth": "350px",
    }
  }
>
  <div
    index={1}
    style={
      Object {
        "MozBoxShadow": "5px 5px 7px #212121",
        "MozTransition": "-moz-transform 0.15s linear",
        "OTransition": "-o-transform 0.15s linear",
        "WebkitBoxShadow": "5px 5px 7px rgba(33, 33, 33, 0.7)",
        "WebkitTransition": "-webkit-transform 0.15s linear",
        "backgroundColor": "aliceblue",
        "boxShadow": "5px 5px 7px rgba(33, 33, 33, 0.7)",
        "color": "#999",
        "display": "block",
        "height": "120px",
        "overflow": "auto",
        "padding": "1em",
        "textDecoration": "none",
        "width": "260px",
      }
    }
  >
    newww text
  </div>
  <div
    className="btn-group"
    style={
      Object {
        "paddingLeft": "25px",
        "paddingTop": "5px",
      }
    }
  >
    <button
      className="btn btn-primary"
      id="editBtn"
      onClick={[Function]}
    >
      Edit
    </button>
    <button
      className="btn btn-danger"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
  <br />
  <br />
  <br />
</div>
`;
